### DISCUSSION TITLE
Strimzi mirrormaker2 consumer offset not getting synced properly.

### DISCUSSION THREAD

**Discussion by sudheesh-87** [strimzi/strimzi-kafka-operator#9849] [General]
I'm testing mirrormaker2 for migrating from old kafka to strimzi kafka , I was testing how the consumer offset is getting synced to target cluster using simple kafka-console-producer and kafka-console-consumer commands , I produced 100 messages to source cluster topic and then consumed 50 messages from source cluster topic , then checked the offset status on both the clusters, the source cluster topic current offset got increased it to 50 as expected,   however the target cluster incremented the offset by only 1 , is there any issue with my configuration that causes this ?  ,  following is my configuration,

Operator - 0.38.0
MM2 version - 3.5.0

<img width="498" alt="Screenshot 2024-03-19 at 9 12 28 AM" src="https://github.com/strimzi/strimzi-kafka-operator/assets/47654598/4930aeff-141e-48fe-95cd-5f0104d73ee1">

And following is the offset status ,

source cluster
-----------
GROUP           TOPIC           PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID     HOST            CLIENT-ID
test-offset     test-mm         9          50              100             50              -               -               -

target strimzi cluster
-------------
GROUP           TOPIC           PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID     HOST            CLIENT-ID
test-offset     test-mm         9          1               100             99              -               -               -

**Comment by scholzj** [comment]
For a start, you should properly format the YAML in your question to make it readable.

**Comment by sudheesh-87** [comment]
@scholzj , I couldn't copy paste the yaml without loosing the format, hence added it as screenshot here ..

**Reply by scholzj** [reply]
Well, that is only part of it. Plus it has some template values inside that make it not that much useful. Also, the offsets are still unformatted. But maybe you should check the `offset.max.lag` option which might have something to do with this? It configures how often is the lag synced.

**Reply by sudheesh-87** [reply]
I've tried to set offset.max.lag : 1 , however still the offset is not synced , following is my running configuration,

![image](https://github.com/strimzi/strimzi-kafka-operator/assets/47654598/6bec7576-398d-4609-bb3b-69c679cd2be2)


Source cluster offset ,

GROUP           TOPIC           PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID     HOST            CLIENT-ID
new             new             0          412             412             0               -               -               -
new             new             1          532             613             81              -               -               -

target cluster offset,

GROUP           TOPIC           PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID     HOST            CLIENT-ID
new             new             0          412             412             0               -               -               -
new             new             1          412             613             201             -               -               -
